pkgs/build-support/prefer-remote-fetch/default.nix

```diff
@@ -0,0 +1,29 @@
+# An overlay that download sources on remote builder.
+# This is useful when the evaluating machine has a slow
+# upload while the builder can fetch faster directly from the source.
+# Usage: Put the following snippet in your usual overlay definition:
+#
+#   self: super:
+#     (super.prefer-remote-fetch self super)
```

Comment:
If you agree on the design, I am going to add this to the nixpkgs manual too.

Reply:
I added a small chapter to the documentation.

Reply:
Did you mean
```
(self: super:
     super.prefer-remote-fetch self super)
```
?

For example, this will work:
```
nixpkgs.overlays = [
(self: super:
     super.prefer-remote-fetch self super)
];
```
But this will not:
```
nixpkgs.overlays = [
self: super:
     (super.prefer-remote-fetch self super)
];
```